[Cocoa] Introduce WKSLinearMediaPlayer

aestes opened the pull request:
#### 4de0f4880b7b12b8f995bfbe480b41503219ab67
<pre>
[Cocoa] Introduce WKSLinearMediaPlayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=268361">https://bugs.webkit.org/show_bug.cgi?id=268361</a>
<a href="https://rdar.apple.com/121907476">rdar://121907476</a>

Reviewed by Jer Noble.

Declared an Objective-C interface for WKSLinearMediaPlayer and a protocol for
WKSLinearMediaPlayerDelegate. The interface is implemented in Swift, where it conforms to the
Playable protocol. The properties exposed to Objective-C are bound to Publishers via key-value
observing, and Playable functions are exposed to Objective-C via the delegate. Since
@_objcImplementation classes can only contain stored properties representable in Objective-C,
Swift-only types are stored indirectly via SwiftOnlyData.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaPlayer.swift: Added.
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift: Drive-by fix to support Public SDK builds on visionOS.
* Source/WebKit/WebKitSwift/LinearMediaKit/WKSLinearMediaPlayer.h: Added.
* Source/WebKit/WebKitSwift/WebKitSwift.h:

Canonical link: <a href="https://commits.webkit.org/273806@main">https://commits.webkit.org/273806@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ab9a948b4448af8c6bb526adac360fb7dd18ec1f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36651 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15591 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38879 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39310 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32853 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18067 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12670 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31455 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37212 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13144 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32412 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11509 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11517 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40556 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33208 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33021 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37453 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11783 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9619 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35570 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13458 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8324 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12193 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12669 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaPlayer.swift:
```diff
@@ -0,0 +1,527 @@
+    var renderingConfiguration: RenderingConfiguration?
+    var thumbnailMaterial: VideoMaterial?
+    var videoMaterial: VideoMaterial?
+    var peculiarEntity: PeculiarEntity?
```
this property name amuses me

rr-codes commented on Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaPlayer.swift:
```diff
@@ -0,0 +1,527 @@
+    }
+
+    public func makeDefaultEntity() -> Entity? {
+        return nil
```
no return

aestes committed:
[Cocoa] Introduce WKSLinearMediaPlayer
https://bugs.webkit.org/show_bug.cgi?id=268361
rdar://121907476

Reviewed by Jer Noble.

Declared an Objective-C interface for WKSLinearMediaPlayer and a protocol for
WKSLinearMediaPlayerDelegate. The interface is implemented in Swift, where it conforms to the
Playable protocol. The properties exposed to Objective-C are bound to Publishers via key-value
observing, and Playable functions are exposed to Objective-C via the delegate. Since
@_objcImplementation classes can only contain stored properties representable in Objective-C,
Swift-only types are stored indirectly via SwiftOnlyData.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaPlayer.swift: Added.
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift: Drive-by fix to support Public SDK builds on visionOS.
* Source/WebKit/WebKitSwift/LinearMediaKit/WKSLinearMediaPlayer.h: Added.
* Source/WebKit/WebKitSwift/WebKitSwift.h:

Canonical link: https://commits.webkit.org/273806@main
